client/src/view/Login.svelte

```diff
@@ -14,7 +34,9 @@
     <span>proximity</span>
   </div>
   <button
-    class="button orange-gradient px-12 py-3 font-medium transition-shadow hover:shadow-2xl hover:shadow-orange-600/50 focus:outline-none focus:ring focus:ring-orange-300 active:opacity-90"
+    class="button orange-gradient px-12 py-3 font-medium transition-shadow hover:shadow-2xl hover:shadow-orange-600/50 focus:outline-none focus:ring focus:ring-orange-300 active:opacity-90 disabled:opacity-40 disabled:shadow-none"
+    on:click={handleSignIn}
+    disabled={buttonDisabled}
```

Comment:
Some indication that a background blocking process is happening in addition to disabling the button will be helpful. Some animation and/or text like "Getting location...", "Logging in..." etc.

Reply:
I agree. I'm planning to add these later.